lib/batch/batch_metadata_updates_655.rb

```diff
@@ -0,0 +1,27 @@
+# Transfer metadata values from Original Publisher to Digital Publisher field
+
+# below is too slow
+#items_needing_transfer = GenericFile.all.reject { |gf| 
+#                           gf.original_publisher == []
+#                         }
+
+solr_items_w_original_publisher_value = ActiveFedora::SolrService.query(
+                                     'original_publisher_tesim:*', 
+                                     { rows: 99999 }
+                                   )
+
+solr_items_w_original_publisher_value.each do |solr_item|
+  # could be either Collection or GenericFile, so we convert the string to
+  # class to get the object by id
+  constant_item = solr_item['has_model_ssim'].first.constantize
+  item = constant_item.find(solr_item['id'])
+
+  item.original_publisher.each do |op|
+    if !item.publisher.include?(op)
+      item.publisher += [op]
+    end
+  end
+
+  item.original_publisher -= item.original_publisher
+  item.save!
+end
```

Comment:
One-time scripts don't *necessarily* warrant tests, but this kind might. It would help:
- to make sure we are not altering something we don't want to
- to get faster feedback on what works and what doesn't
- to convince me this works (don't know the structure of the index, so don't know if this works well)

I would emphasize the 2nd point: testing this script on a complete index is very time consuming. Not only for the script, but for you: you need to reload the full index if you made a mistake (you maybe are running this on a remote machine which adds time...) Using a smaller one might help you iterate faster.

Reply:
Sure, I agree we should test when needed. But maybe answering some concerns may prove it's not necessary for this one. 

- I know we won't be altering the wrong items, because Original Publisher was custom added to only the two; Collection and GenericFile
- It's not needed to know the structure of the index. We only query it because it's much faster than ActiveFedora(these are not DB objects, but Fedora objects) to get the objects we want to update. 
- I have been testing this manually on my local machine(for what it's worth), so testing against my development index works. 

If these are still not convincing enough, I'll look into how to test this.

Reply:
In pragmatic terms, my concern is that using the full index/db as the basis to iterate development might be too slow.
Given that DigitalHub is in maintenance mode, it is not too bad if we don't have a test index. I just want to highlight the potential stakes:

Unless you always get it right on the first pass of your code, I only see how slow and painful the experience must be: write code, run on db/index, oups doesn't work, reload full db and index, write code, oups, full reload... 
This is also why I am wondering how slow/fast is the running code (order of magnitude), because that will also hamper this flow. Finally, this is at least the second time something like this is run on DigitalHub. Having a nice small setup for this would save time every one of those instances.

Just something to keep in mind in the future. You can merge this.

Reply:
Ahh, I understand now. Yes, I can see how this can be a headache without tests, and I see the value now. As you stated earlier, one time scripts don't always need it, and because of that it has never crossed my mind. But I will definitely try that route for the next one(probably as soon as the next PR). Thanks for raising these concerns.